### DISCUSSION TITLE
Add more options

### DISCUSSION THREAD

**Discussion by execonic** [RobertWeinmeister/dokuwiki-mermaid#19] [Ideas]
This plugin is fantastic, thanks for your contribution! 💯 

I have some thoughts about this plugin. 

- Move `raw` to the `<mermaid>` tag. i.e `<mermaid raw>`.
- Add a new option `css`, e.g. `css=custom`, to specify the CSS to be applied to the container so that we can customize the style of the container without adding more options.
- Add a **global setting** to specify whether or not raw code is passed to  Mermaid by default. Also, I think an opposite option for `raw` is needed. How about `raw=true` or `raw=false`?
-  Add a **global setting** to customize CDN.
- Add a **global setting** to customize [theme](https://mermaid.js.org/config/theming.html).

Thanks again!

**Comment by RobertWeinmeister** [comment]
Thanks for these ideas.

1. The current implementation was chosen to avoid compatibility problems with already written Mermaid tags. It should also not make a big difference.
2. I thought this is already possible by using a DokuWiki template. The plugin itself uses CSS to style the generated mermaid div.
3. At the moment, I see the raw mode as the exception, as the whole point is the integration with DokuWiki functionality. I would like to keep its usage to a minimum. Also, if a user would ever change the global default, all Mermaid code would have to be changed too.
4. Could be done, but I do not see a big need for this.
5. I will look into this.

**Comment by RobertWeinmeister** [comment]
The five default themes are available via the [plugin configuration](https://github.com/RobertWeinmeister/dokuwiki-mermaid?tab=readme-ov-file#default-theme).
